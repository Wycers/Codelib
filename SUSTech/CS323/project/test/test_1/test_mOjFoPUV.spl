
struct Info{
    char name[30];
    int age;
    struct address{
        char area_name[39];
        int house_no;
        char district[39];
    } address;
};
int f(){
  int a = 0;
}