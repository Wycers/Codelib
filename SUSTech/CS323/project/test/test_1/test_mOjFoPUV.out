Program (2)
  ExtDefList (2)
    ExtDef (2)
      Specifier (2)
        StructSpecifier (2)
          STRUCT
          ID: Info
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: char
              DecList (3)
                Dec (3)
                  VarDec (3)
                    VarDec (3)
                      ID: name
                    LB
                    INT: 30
                    RB
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: int
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: age
                SEMI
              DefList (5)
                Def (5)
                  Specifier (5)
                    StructSpecifier (5)
                      STRUCT
                      ID: address
                      LC
                      DefList (6)
                        Def (6)
                          Specifier (6)
                            TYPE: char
                          DecList (6)
                            Dec (6)
                              VarDec (6)
                                VarDec (6)
                                  ID: area_name
                                LB
                                INT: 39
                                RB
                          SEMI
                        DefList (7)
                          Def (7)
                            Specifier (7)
                              TYPE: int
                            DecList (7)
                              Dec (7)
                                VarDec (7)
                                  ID: house_no
                            SEMI
                          DefList (8)
                            Def (8)
                              Specifier (8)
                                TYPE: char
                              DecList (8)
                                Dec (8)
                                  VarDec (8)
                                    VarDec (8)
                                      ID: district
                                    LB
                                    INT: 39
                                    RB
                              SEMI
                      RC
                  DecList (9)
                    Dec (9)
                      VarDec (9)
                        ID: address
                  SEMI
          RC
      SEMI
    ExtDefList (11)
      ExtDef (11)
        Specifier (11)
          TYPE: int
        FunDec (11)
          ID: f
          LP
          RP
        CompSt (11)
          LC
          DefList (12)
            Def (12)
              Specifier (12)
                TYPE: int
              DecList (12)
                Dec (12)
                  VarDec (12)
                    ID: a
                  ASSIGN
                  Exp (12)
                    INT: 0
              SEMI
          RC
